src/material-experimental/mdc-list/list-base.ts

```diff
@@ -77,20 +86,113 @@ export abstract class MatListItemBase implements AfterContentInit, OnDestroy, Ri
     this._ngZone.runOutsideAngular(() => {
       this._subscriptions.add(this.lines.changes.pipe(startWith(this.lines))
           .subscribe((lines: QueryList<ElementRef<Element>>) => {
-            this._element.nativeElement.classList
+            this._element.classList
                 .toggle('mat-mdc-list-item-single-line', lines.length <= 1);
             lines.forEach((line: ElementRef<Element>, index: number) => {
               toggleClass(line.nativeElement,
                   'mdc-list-item__primary-text', index === 0 && lines.length > 1);
               toggleClass(line.nativeElement, 'mdc-list-item__secondary-text', index !== 0);
             });
-            setLines(lines, this._element, 'mat-mdc');
+            setLines(lines, this._elementRef, 'mat-mdc');
           }));
     });
   }
+}
+
+@Directive()
+/** @docs-private */
+export abstract class MatListBase {
+  @HostBinding('class.mdc-list--non-interactive')
+  _isNonInteractive: boolean = true;
+}
+
+@Directive()
+export abstract class MatInteractiveListBase extends MatListBase
+    implements AfterViewInit, OnDestroy {
+  @HostListener('keydown', ['$event'])
+  _handleKeydown(event: KeyboardEvent) {
+    const index = this._indexForElement(event.target as HTMLElement);
+    this._foundation.handleKeydown(
+        event, this._itemAtIndex(index)._element === event.target, index);
+  }
+
+  @HostListener('click', ['$event'])
+  _handleClick(event: MouseEvent) {
+    this._foundation.handleClick(this._indexForElement(event.target as HTMLElement), false);
+  }
+
+  @HostListener('focusin', ['$event'])
+  _handleFocusin(event: FocusEvent) {
+    this._foundation.handleFocusIn(event, this._indexForElement(event.target as HTMLElement));
+  }
+
+  @HostListener('focusout', ['$event'])
+  _handleFocusout(event: FocusEvent) {
+    this._foundation.handleFocusOut(event, this._indexForElement(event.target as HTMLElement));
+  }
+
+  @ContentChildren(MatListItemBase) _items: QueryList<MatListItemBase>;
+
+  protected _adapter: MDCListAdapter = {
+    getListItemCount: () => this._items.length,
+    listItemAtIndexHasClass:
+        (index, className) => this._itemAtIndex(index)._element.classList.contains(className),
+    addClassForElementIndex:
+        (index, className) => this._itemAtIndex(index)._element.classList.add(className),
+    removeClassForElementIndex:
+        (index, className) => this._itemAtIndex(index)._element.classList.remove(className),
+    getAttributeForElementIndex:
+        (index, attr) => this._itemAtIndex(index)._element.getAttribute(attr),
+    setAttributeForElementIndex:
+        (index, attr, value) => this._itemAtIndex(index)._element.setAttribute(attr, value),
+    setTabIndexForListItemChildren:
+        (index, value) => this._itemAtIndex(index)._setTabIndexForChildren(Number(value)),
+    getFocusedElementIndex: () => this._indexForElement(this._document?.activeELement),
```

Comment:
```suggestion
    getFocusedElementIndex: () => this._indexForElement(this._document?.activeElement),
```

It's a bit odd that we cannot use `Document` as type for the injection. That is because the type will be preserved in the metadata output and _could_ cause SSR issues. An idea would be to not assign it directly in the constructor, but assign it to a variable manually that is properly typed.

Reply:
It would be nice if that worked, I do hate how we have `document` typed as `any` all over our code

Reply:
There is a way to get this typed. i.e. assigning the `any` document manually to a class member typed with `Document`.

We could potentially fix this now that we have the metadata/decorator transform in NGC. We'd just need to omit the type in the output if there is an `@Inject`.

Reply:
I used the workaround you suggested for now